### PR TITLE
ENCD-6262-elements-cloning-and-elements-mappings-display

### PR DESCRIPTION
**Pull request by forresttanaka**
* I removed the `analysisObjects` collapsed-state initialization as it never seemed to come up with results, and the rules for the default collapse/expanded states of the various file tables has changed in this ticket anyway.
* Related, I removed the collapsed states from the raw data table and the raw sequencing table so that FileTable can coordinate all the tables for the new collapse rules.

### CODE REVIEW

**Comment by forresttanaka** [comment]
~https://encd-6262-fcs-files-pv2-fytanaka.demo.encodedcc.org/search/?type=FunctionalCharacterizationSeries&related_datasets.elements_cloning=*&limit=all~

No `elements_mappings` exist in production data. I have a couple in local test data.

File: src/encoded/static/components/dataset.js [hunk 1]
```diff
@@ -2868,12 +2870,15 @@ export const SeriesComponent = ({
                     supplementalShortLabels={supplementalShortLabels}
                     showReplicateNumber={false}
                     hideControls={!METADATA_SERIES_TYPES.includes(context['@type'][0])}
-                    collapseNone
```
Comment (emro) [inline src/encoded/static/components/dataset.js:2871]: This is funny, I guess this isn't used anywhere. Maybe it's left from some previous change that did need it.

File: src/encoded/static/components/dataset.js [hunk 2]
```diff
@@ -3164,13 +3173,119 @@ DiseaseSeries.contextTypes = {
...
+    datasets.map((dataset) => {
+        let assemblies = dataset.assembly || [];
+        let files = [];
+        if (dataset.analyses?.length > 0) {
+            // Get the file objects for the dataset's analyses' files.
+            const analysisFilesPaths = dataset.analyses.reduce((accFiles, analysis) => accFiles.concat(analysis.files), []);
+            const cloningAndAnalysisFilePaths = _.intersection(dataset.files, analysisFilesPaths);
+            files = cloningAndAnalysisFilePaths
+                .map((filePath) => allFiles.find((file) => file['@id'] === filePath))
+                .filter((file) => file);
```
Comment (emro) [inline src/encoded/static/components/dataset.js:3195]: What does this do? (`.filter((file) => file)`) I don't know if I've seen this before.

**Review by emro — CHANGES_REQUESTED** [review]
The code all looks good but the conflicting file needs to be resolved.

File: src/encoded/static/components/dataset.js (hunk 2 above)
Comment (forresttanaka) [reply src/encoded/static/components/dataset.js:3195]: It’s a local version of _.compact(). It just removes any falsy entries in the array, and easier than wrapping the whole RHS in _.compact().

File: src/encoded/static/components/dataset.js (hunk 1 above)
Comment (forresttanaka) [reply src/encoded/static/components/dataset.js:2871]: The rules about collapsing have changed in this ticket, so that’s why this went away.

**Comment by forresttanaka** [comment]
I updated https://encd-6262-fcs-files-qa1-fytanaka.demo.encodedcc.org/ after rebasing.

**Review by emro — DISMISSED** [review]
Looks good!

File: src/encoded/static/components/batch_download/cloning_mappings.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        super(dataset, new QueryString());
+    }
+
+    preBuildDownloadOptions(dataset) {
+        this._hasAnalyses = !!(dataset.analyses?.length > 0);
+        this._downloadOptionsTemplate = [
+            {
+                id: 'all-files',
+                label: 'Download files',
+                title: `All experiment${this._hasAnalyses ? `${uc.rsquo}s analyses` : ''} files`,
```
Comment (emro) [inline src/encoded/static/components/batch_download/cloning_mappings.js:26]: Should this be "all experiments' analysis files"?
Screen Shot 2022-02-03 at 10 39 18 AM

**Review by emro — DISMISSED** [review]
Maybe one re-wording thing, but code all looks great.

**Review by emro — APPROVED** [review]
Code changes look good. This is a very complicated ticket.

**Pull request merged**